strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -191,6 +299,8 @@ def mbsqueeze(self, k, r, phi, r_anc, eta_anc, avg):
             X2, Y2 = self.expandXY([k], X, Y)
             self.apply_channel(X2, Y2)
 
+        # Add new ancilla mode, interfere it and measure it
+        # Delete ancilla mode from active list
         if not avg:
             self.add_mode()
```

Comment:
is it possible to the map without having to add the extra mode?

Reply:
We need to sample a value for the homodyne measurement, so this seems like a straightforward way. The extra mode is Gaussian, so it doesn't increase the number of weights, so I don't think it is too memory intensive.